l2geth/rollup/sync_service.go

```diff
@@ -1151,13 +1157,86 @@ func (s *SyncService) verifyFee(tx *types.Transaction) error {
 	return nil
 }
 
-func (s *SyncService) ValidateSequencerTransaction(tx *types.Transaction, sequencer common.Address) error {
+func (s *SyncService) VerifiedTxCount() (uint64, error) {
+	var pendingTxCount uint64
+	pendingTxs, err := s.txpool.Pending()
+	if err != nil {
+		return 0, err
+	}
+	for _, txs := range pendingTxs {
+		for _, tx := range txs {
+			if err := s.ValidateSequencerTransaction(tx); err == nil {
+				pendingTxCount++
+			}
+		}
+	}
+	return pendingTxCount, nil
+}
+
+// ApplyUpdateGasPriceTxs apply tx for update gasPrice
+func (s *SyncService) ApplyUpdateGasPriceTxs() {
+	for {
+		if len(s.updateGasPriceTxPool) == 0 {
+			break
+		}
+		tx := <-s.updateGasPriceTxPool
+		if err := s.ValidateAndApplySequencerTransaction(tx, &types.BatchTxSetProof{}); err != nil {
+			log.Error("apply update gasPrice tx error", "err_msg", err.Error())
+		}
+	}
+}
+
+func (s *SyncService) IsUpdateGasPriceTx(tx *types.Transaction) bool {
+	if tx == nil {
+		return false
+	}
+	from, err := types.Sender(s.signer, tx)
+	if err != nil {
+		return false
+	}
+	gpoOwner := s.GasPriceOracleOwnerAddress()
```

Comment:
How about `gpoOwner == nil` , return true directly？

Reply:
The Owner is allowed to be empty when the contract is not deployed